numba/tests/test_random.py

```diff
@@ -1291,6 +1294,129 @@ def test_multinomial_3_tuple(self):
+        res = cfunc(alpha, size)
+        self._check_sample(alpha, size, res)
+        # A case with noninteger probabilities
+        alpha = np.array([1, 1, 1.5, 1], dtype=np.float64)
+        res = cfunc(alpha, size)
+        self._check_sample(alpha, size, res)
+        alpha = tuple(alpha)
+        res = cfunc(alpha, size)
+        self._check_sample(alpha, size, res)
+
+    def test_dirichlet_int(self):
+        """
+        Test dirichlet(alpha, size=int)
+        """
+        cfunc = jit(nopython=True)(numpy_dirichlet)
+        alpha = self.alpha
+        size=10
+        res = cfunc(alpha, size)
+        self._check_sample(alpha, size, res)
+        # pvals as tuple
+        pvals = tuple(alpha)
+        res = cfunc(alpha, size)
+        self._check_sample(alpha, size, res)
+        # A case with extreme probabilities
+        alpha = np.array([1, 1, 10000, 1], dtype=np.float64)
+        res = cfunc(alpha, size)
+        self._check_sample(alpha, size, res)
+        # A case with noninteger probabilities
+        alpha = np.array([1, 1, 1.5, 1], dtype=np.float64)
+        res = cfunc(alpha, size)
+        self._check_sample(alpha, size, res)
+        alpha = tuple(alpha)
+        res = cfunc(alpha, size)
+        self._check_sample(alpha, size, res)
+
+    def test_dirichlet_tuple(self):
+        """
+        Test dirichlet(alpha, size=tuple)
+        """
+        cfunc = jit(nopython=True)(numpy_dirichlet)
+        alpha = self.alpha
+        size=(10,10)
+        res = cfunc(alpha, size)
+        self._check_sample(alpha, size, res)
+        # pvals as tuple
+        pvals = tuple(alpha)
+        res = cfunc(alpha, size)
+        self._check_sample(alpha, size, res)
+        # A case with extreme probabilities
+        alpha = np.array([1, 1, 10000, 1], dtype=np.float64)
+        res = cfunc(alpha, size)
+        self._check_sample(alpha, size, res)
+        # A case with noninteger probabilities
+        alpha = np.array([1, 1, 1.5, 1], dtype=np.float64)
+        res = cfunc(alpha, size)
+        self._check_sample(alpha, size, res)
+        alpha = tuple(alpha)
+        res = cfunc(alpha, size)
+        self._check_sample(alpha, size, res)
+
+    def test_dirichlet_exceptions(self):
+        cfunc = jit(nopython=True)(numpy_dirichlet)
+        alpha = tuple((0,1,1))
+        self.assertRaises(ValueError, cfunc, alpha, 1)
+        alpha = self.alpha
+        size = True
+        with self.assertTypingError():
+            cfunc(alpha, size)
+        size = 3j
+        with self.assertTypingError():
+            cfunc(alpha, size)
+        size = 1.5
+        with self.assertTypingError():
+            cfunc(alpha, size)
+        size = (1.5, 1)
+        with self.assertTypingError():
+            cfunc(alpha, size)
+        size = (3j, 1)
+        with self.assertTypingError():
+            cfunc(alpha, size)
+        size = (3j, 3j)
+        with self.assertTypingError():
+            cfunc(alpha, size)
```

Comment:
Whilst not in use in this test module (as it's relatively old), Numba uses this convention for checking that errors (particularly typing errors) come from an expected place:
https://github.com/numba/numba/blob/e395e00cd7720494f68b44e97dc3e51b1a0c9757/numba/tests/test_np_functions.py#L600-L605
essentially, the error message strings are captured and checked to make sure they are as expected.

Reply:
Great, I added this in and cleaned up that set of tests.